tests/tasks/test_tasks.py

```diff
@@ -0,0 +1,90 @@
+from time import sleep
+import json
+import requests
+
+from task_bodies import (
+    output_dir,
+    uppercase_task_body,
+    decryption_task_body,
+    uppercase_task_with_decryption_body
+)
+
+tes_url = "http://localhost:8090/ga4gh/tes/v1"
+headers = {"accept": "application/json", "Content-Type": "application/json"}
+WAIT_STATUSES = ("UNKNOWN", "INITIALIZING", "RUNNING")
+
+
+def create_task(tasks_body):
+    return requests.post(
+        url=f"{tes_url}/tasks", headers=headers, json=tasks_body
+    )
+
+
+def get_task(task_id):
+    return requests.get(
+        url=f"{tes_url}/tasks/{task_id}", headers=headers
+    )
+
+
+def get_task_state(task_id):
+    def wait_for_task_completion():
+        nonlocal task_state
+        get_response = get_task(task_id)
+        task_state = json.loads(get_response.text)["state"]
+        while task_state in WAIT_STATUSES:
+            sleep(0.5)
+            get_response = get_task(task_id)
+            task_state = json.loads(get_response.text)["state"]
+
+    task_state = ""
+    wait_for_task_completion()
+    return task_state
+
+
+def test_uppercase_task():
+    """Test task that outputs uppercase version of input"""
+
+    post_response = create_task(uppercase_task_body)
+    assert post_response.status_code == 200
```

Comment:
Consider using `pytest` for running tests. Then, consider defining `pytest` fixtures to simplify the setup stages for your tests to minimize repetition.

I have marked these two lines, but others repeat as well, so possibly those could be included as well in a fixture (or consider having multiple fixtures).